sql/core/src/test/resources/sql-tests/inputs/udf/udf-pivot.sql

```diff
@@ -0,0 +1,307 @@
+-- This test file was converted from pivot.sql.
+
+--Note some test cases have been commented as the current integrated UDFs cannot handle complex types
+
+create temporary view courseSales as select * from values
+  ("dotNET", 2012, 10000),
+  ("Java", 2012, 20000),
+  ("dotNET", 2012, 5000),
+  ("dotNET", 2013, 48000),
+  ("Java", 2013, 30000)
+  as courseSales(course, year, earnings);
+
+create temporary view years as select * from values
+  (2012, 1),
+  (2013, 2)
+  as years(y, s);
+
+create temporary view yearsWithComplexTypes as select * from values
+  (2012, array(1, 1), map('1', 1), struct(1, 'a')),
+  (2013, array(2, 2), map('2', 2), struct(2, 'b'))
+  as yearsWithComplexTypes(y, a, m, s);
+
+-- pivot courses
+SELECT * FROM (
+  SELECT udf(year), course, earnings FROM courseSales
+)
+PIVOT (
+  udf(sum(earnings))
+  FOR course IN ('dotNET', 'Java')
```

Comment:
I can test it by myself but I thought you know the results - just out of curiosity,  what do we get if we do `FOR udf(course)`?

Reply:
`FOR udf(course)` in line 29 will result in a `org.apache.spark.sql.catalyst.parser.ParseException`